cpp/src/io/parquet/reader_impl.cu

```diff
@@ -854,7 +855,11 @@ void reader::impl::read_column_chunks(
     if (io_size != 0) {
       auto& source = _sources[chunk_source_map[chunk]];
       if (source->is_device_read_preferred(io_size)) {
-        page_data[chunk] = source->device_read(io_offset, io_size, stream);
+        auto buffer        = rmm::device_buffer(io_size, stream);
+        auto fut_read_size = source->device_read_async(
+          io_offset, io_size, static_cast<uint8_t*>(buffer.data()), stream);
+        fut_read_tasks.emplace_back(std::move(fut_read_size));
+        page_data[chunk] = datasource::buffer::create(std::move(buffer));
```

Comment:
I would prefer fewer tmp variables, but it's probably just a personal preference.
```suggestion
          page_data[chunk] = rmm::device_buffer(io_size, stream);
          fut_read_tasks.emplace_back(source->device_read_async(
            io_offset, io_size, static_cast<uint8_t*>(page_data[chunk].data()), stream));
```

Reply:
`page_data[chunk]` is not `rmm::device_buffer`, it's a `datasource::buffer` and the `data()` method of `buffer` returns a `uint8_t const*`. So I suppose once it's a buffer, it's immutable?